lib/stitch_plan/lock_stitch.py

```diff
@@ -0,0 +1,202 @@
+import re
+from copy import copy
+from math import degrees
+
+from inkex import DirectedLineSegment, Path
+from shapely.geometry import LineString
+from shapely.ops import substring
+
+from ..i18n import _
+from ..svg import PIXELS_PER_MM
+from ..utils import string_to_floats
+from .stitch import Stitch
+
+
+class LockType:
+    def __init__(self, lock_id=None, name=None, path=None, path_type=None,
+                 scale_percent=100, scale_absolute=0.7):
+
+        self.id: str = lock_id
+        self.name: str = name
+        self.path: str = path
+        self.path_type: str = path_type
+        self.scale_percent: float = scale_percent
+        self.scale_absolute: float = scale_absolute
+
+    def __repr__(self):
+        return "LockType(%s, %s, %s, %s, %s, %s)" % (self.id, self.name, self.path, self.path_type,
+                                                     self.scale_percent, self.scale_absolute)
+
+    def __str__(self):
+        return str(self.id) or ""
+
+    def copy(self, scale_percent=None, scale_absolute=None):
+        cp = copy(self)
+        cp.set('scale_percent', scale_percent or self.scale_percent)
+        cp.set('scale_absolute', scale_absolute or self.scale_absolute)
+        return cp
+
+    def set(self, attribute, value):
+        setattr(self, attribute, value)
```

Comment:
It's reasonable to let users of this class set the path directly, like `lock_start.path = self.lock_custom_start`.  If you're worried they might set something invalid, you can use a property, like in the ColorBlock class (lib/stitch_plan/color_block.py).  You could even do the _check_custom() call right in the property setter function, and use the fallback instead of what they passed if it fails validation.